src/Components/Organisms/Navigation/navigation.tsx

```diff
@@ -0,0 +1,102 @@
+import "../Navigation/navigation.scss";
+import { NavigationData, Cart, MenuData, backButtonData } from "../../../data";
+import Haburger from "../../Attoms/Hamburger-button/hamburger";
+import CartDropDown from "../../Molecules/cart-drop-down/cart-drop-down";
+import Menu from "../../Molecules/menu/menu";
+import SubMenu from "../../Molecules/sub-menu/sub-menu";
+import { useRef, useState, useEffect } from "react";
+
+const Navigation = (
+  props: NavigationData & Cart & MenuData & backButtonData
+) => {
+  const [isCartOpen, setCartOpen] = useState(false);
+  const [isMainMenuOpen, setMainMenuOpen] = useState(false);
+  const [isSubMenuOpen, setSubMenuOpen] = useState(false);
```

Comment:
optional, `useState` accepts generic type, can explicitly pass the type you want or let TS infer it
`const [isCartOpen, setCartOpen] = useState<boolean>(false);`